packages/fuels-core/src/types/param_types.rs

```diff
@@ -67,7 +67,8 @@ impl ParamType {
         param_type: &ParamType,
         available_bytes: usize,
     ) -> Result<usize> {
-        let memory_size = param_type.compute_encoding_width() * WORD_SIZE;
+        let encoding_width = param_type.compute_encoding_width()?;
+        let memory_size = encoding_width.checked_mul(WORD_SIZE).ok_or_else(|| error!(InvalidData, "attempt to multiply encoding_width ({encoding_width:?}) by WORD_SIZE ({WORD_SIZE:?}) with overflow"))?;
```

Comment:
same remark here regarding the error message that gives the user too much information.